### ISSUE TITLE
OWLOntologyManagerImpl.getVersions() calls get() on Optional without checking if the contained value is present

### ISSUE THREAD

**Issue by matthewhorridge** [owlcs/owlapi#401]
Here is the code

```java
Set<OWLOntology> onts = new HashSet<>();
for (OWLOntology ont : getOntologies()) {
    if (ontology.equals(ont.getOntologyID().getOntologyIRI().get())) {
        onts.add(ont);
    }
}
return onts;
```

**Comment by ignazio1977** [comment]
I've got the impression that code is wrong twice. Apart from the get without isPresent, it is also comparing an OWLOntology for equality with a String.
Should be:
`if(ontology.getOntologyID().getOntologyIRI().equals(ont.getOntologyID().getOntologyIRI())`

**Comment by matthewhorridge** [comment]
Yes, indeed.

**Comment by ignazio1977** [comment]
Fix copied to version 4.0.2

**Comment by ignazio1977** [comment]
Bug does not exist in version 5
